### ISSUE TITLE
False positive for: list((1,2))+[3,]

### ISSUE THREAD

**Issue by colinRawlings** [microsoft/pylance-release#487]
## Environment data

-   Language Server version: 2020.10.0
-   OS and version: Win 10 Build 20175
-   Python version (& distribution if applicable, e.g. Anaconda): Python.org 3.6.8

## Expected behaviour

No red squiggle under the expression: `list((1,2))+[3,]`
## Actual behaviour

Red squiggle with message:
```
 Operator "+" not supported for types "list[Literal[1, 2]]" and "List[int]"
 Operator "+" not supported for types "list[Literal[1, 2]]" and "List[int]"Pylance 
(reportGeneralTypeIssues)
```

Also not sure why the message is duplicated?

## Logs

```
XXX
```

## Code Snippet / Additional information

```python
XXX
```

**Comment by jakebailey** [comment]
I think this is somewhat expected; the first list is inferred to be `List[Literal[1, 2]]` and that isn't compatible with the other type.

If you write the following, it type checks:

```python
x: List[int] = list((1,2))
y = x + [3,]
```

**Comment by erictraut** [comment]
@jakebailey, I'll take a deeper look at this one. The expression `(1,2)` is inferred to be `Tuple[Literal[1], Literal[2]]`, and I think that's correct. But there's a question (not defined in any of the PEPs) about how a tuple with heterogeneous entry types should be translated into an `Iterable`. Pyright is currently unioning all of the individual types, which is why it results in an `Iterable[Literal[1, 2]]` in this case. But arguably, literals should be stripped during this conversion, resulting to `Iterable[int]`. That would then produce a `List[int]` when passed through the `list` constructor. I need to investigate more to see if this would break other things.

**Comment by erictraut** [comment]
I've improved the bidirectional type inference for tuple expressions. The logic is now smarter about when to honor literals and when to strip them.

This will be included in the next release of pylance.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.10.2, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#2020102-21-october-2020
